python/lsst/ctrl/oods/butlerAttendant.py

```diff
@@ -105,11 +106,15 @@ async def ingest(self, file_list):
 
         # Ingest images.
         await asyncio.sleep(0)
+        new_list = file_list
+        if self.s3profile:
+            # rewrite URI to add s3profile
+            new_list = [s.replace(netloc=f"{self.s3profile}@{s.netloc}") for s in file_list]
```

Comment:
Is there a risk that a profile is already present and you would need to check for `@` in `s.netloc`?

Reply:
Before I did any of this, I asked whether or not the profile could be added when the file was initially deposited so I wouldn't have to deal with it as above, but the answer was no.